### DISCUSSION TITLE
Support request: Orphaned ZSH Processes

### DISCUSSION THREAD

**Discussion by Kaelten** [aws/codewhisperer-command-line-discussions#96] [Support ticket]
### Checks

- [X] I ran `cw doctor` in the affected command line session and it didn't resolve my issue
- [X] I ran `cw restart` and it didn't resolve my issue
- [X] I searched [the discussions in this repo](https://github.com/aws/codewhisperer-command-line-discussions/discussions?q=) and there are no duplicates of my issue

### Expected behaviour

I do not expect CW to leave orphaned ZSH processes on my mac until I reboot.

### Actual behaviour

I would have dozens or even hundreds of zsh processes on my machine.  Triggering auto completes in iterm2 with code whisperer caused these to happen and reverting back to using fig instead eliminated this and other issues.

The processes look like this in ps:
`/bin/zsh -lic for key in ${(k)commands}; do echo $key; done && alias +r`

**Comment by grant0417** [comment]
That command should list the alises you have, can you try running it in your terminal and see if it exits? `/bin/zsh -lic 'for key in ${(k)commands}; do echo $key; done && alias +r'`

**Reply by Kaelten** [reply]
I tried running it manually and it seems that the command was triggering a new terminal session that could include an oh-my-zsh update check and it hung on my first run to ask if I wanted to udpate.  I'm not sure that that woudl explain the issue completely or not, but during this same time frame opening a new iterm2 window/session would not trigger this oh-my-zsh update question.

**Reply by grant0417** [reply]
That would explain it, I will add a timeout to ensure the processes don't live forever.

**Comment by jmcvea** [comment]
I'm seeing the same behavior where 2200+ processes for `/bin/zsh -lic 'for key in ${(k)commands}; do echo $key; done && alias +r'` were reported as running.  I've killed them all now, but some of them were started 4 days ago and still existed.  

When I run `/bin/zsh -lic 'for key in ${(k)commands}; do echo $key; done && alias +r'` directly, I don't get prompted for any updates like OP did.

In my shell configuration, I also use [atuin](https://github.com/atuinsh/atuin).  When I disable atuin, then I don't have any problems with  an accumulation of these processes.

Should the timeout implementation be included in the 1.2.0 release?  Any thoughts on interaction or behavior between atuin and Q?

| Configuration | version |
|--------|--------|
| zsh | zsh 5.9 (x86_64-apple-darwin23.0) |
|[Zim](https://zimfw.sh/) |  v1.13.1 |
|Q for command line | v1.2.0 |
|[atuin](https://github.com/atuinsh/atuin) | v18.3.0 |

**Reply by brandonskiser** [reply]
The timeout implementation is included in the 1.2.0 release. I've tried using atuin and Q together but am not able to reproduce the issue, nothing immediately jumps out that could be causing the issue. You have some steps you've taken to reproduce?
